### PR TITLE
Add logic for edit and list feature for diet goals

### LINKED ISSUE #28: Add logic for editing goals
We have created to goal class. We need to link between user input and editing goals. Extension of issue #12.

### PR DESCRIPTION
**Pull request by yicheng-toh**
Users can now edit and list the target values for the goals.

Fixes #28, #36

### CODE REVIEW

File: src/main/java/athleticli/commands/diet/EditDietGoalCommand.java [hunk 1]
```diff
@@ -2,20 +2,71 @@
...
+            if (!isNutrientGoalInCurrentDietGoalList) {
+                throw new AthletiException(String.format(Message.MESSAGE_DIETGOAL_NOT_EXISTED, userNewNutrient));
+            }
+        }
+
+        // Edit updated goals to current diet goals
+        int newTargetValue;
+        for(DietGoal userUpdatedDietGoal : userUpdatedDietGoals){
+            for(DietGoal currentDietGoal: currentDietGoals){
+                if (!userUpdatedDietGoal.getNutrients().equals(currentDietGoal.getNutrients())){
```
Comment (skylee03) [inline src/main/java/athleticli/commands/diet/EditDietGoalCommand.java:61]:
```suggestion
        for (DietGoal userUpdatedDietGoal : userUpdatedDietGoals) {
            for (DietGoal currentDietGoal: currentDietGoals) {
                if (!userUpdatedDietGoal.getNutrients().equals(currentDietGoal.getNutrients())) {
```

File: src/test/java/athleticli/commands/diet/EditDietGoalCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+    @Test
+    void execute_emptyInputList_expectCorrectMessage() {
+        try {
+            EditDietGoalCommand editDietGoalCommand = new EditDietGoalCommand(emptyInputDietGoals);
+            String[] expectedString = {"These are your goals:\n", ""};
+            String[] actualString = editDietGoalCommand.execute(data);
+            assertArrayEquals(expectedString, actualString);
+        } catch (AthletiException e) {
+            assert (false);
```
Comment (skylee03) [inline src/test/java/athleticli/commands/diet/EditDietGoalCommandTest.java:52]: Maybe `fail()` would be more appropriate here?

**Review by skylee03 — COMMENTED** [review]
LGTM. I think using `assert(false)` should be avoided. Maybe `fail()` would be more appropriate here?

File: src/test/java/athleticli/commands/diet/EditDietGoalCommandTest.java (hunk 2 above)
Comment (yicheng-toh) [reply src/test/java/athleticli/commands/diet/EditDietGoalCommandTest.java:52]: Ok I will change them to fail().

**Review by skylee03 — COMMENTED** [review]
LGTM. `fail` is a method, and thus there should be no spaces before the `(`.

**Review by skylee03 — APPROVED** [review]
LGTM!

**Pull request merged**